src/lib/vehicles.json

```diff
@@ -2915,6 +2943,64 @@
       ]
     }
   ],
+  "Tesla": [
+    {
+      "name": "Tesla Model 3 (with HW3) 2019-23",
+      "model": "Model 3 (with HW3)",
+      "years": "2019-23",
+      "year_list": "2019, 2020, 2021, 2022, 2023",
+      "package": "All",
+      "angled_mount": false,
+      "video_link": "",
+      "harness_connector": "Tesla A",
+      "detail_sentence": "openpilot upgrades your <strong>Tesla Model 3 (with HW3)</strong> with automated lane centering <strong>at all speeds</strong>, and adaptive cruise control <strong>that automatically resumes from a stop</strong>.",
+      "footnotes": [
+        "To verify which hardware type your vehicle has, refer to \u003ca href=\"https://www.notateslaapp.com/news/2173/how-to-check-if-your-tesla-has-hardware-4-ai4-or-hardware-3\"\u003ethis documentation\u003c/a\u003e."
+      ]
+    },
+    {
+      "name": "Tesla Model 3 (with HW4) 2024",
+      "model": "Model 3 (with HW4)",
+      "years": "2024",
+      "year_list": "2024",
+      "package": "All",
+      "angled_mount": false,
+      "video_link": "",
+      "harness_connector": "Tesla B",
+      "detail_sentence": "openpilot upgrades your <strong>Tesla Model 3 (with HW4)</strong> with automated lane centering <strong>at all speeds</strong>, and adaptive cruise control <strong>that automatically resumes from a stop</strong>.",
+      "footnotes": [
+        "To verify which hardware type your vehicle has, refer to \u003ca href=\"https://www.notateslaapp.com/news/2173/how-to-check-if-your-tesla-has-hardware-4-ai4-or-hardware-3\"\u003ethis documentation\u003c/a\u003e."
+      ]
+    },
+    {
+      "name": "Tesla Model Y (with HW3) 2020-23",
+      "model": "Model Y (with HW3)",
+      "years": "2020-23",
+      "year_list": "2020, 2021, 2022, 2023",
+      "package": "All",
+      "angled_mount": false,
+      "video_link": "",
+      "harness_connector": "Tesla A",
+      "detail_sentence": "openpilot upgrades your <strong>Tesla Model Y (with HW3)</strong> with automated lane centering <strong>at all speeds</strong>, and adaptive cruise control <strong>that automatically resumes from a stop</strong>.",
+      "footnotes": [
+        "To verify which hardware type your vehicle has, refer to \u003ca href=\"https://www.notateslaapp.com/news/2173/how-to-check-if-your-tesla-has-hardware-4-ai4-or-hardware-3\"\u003ethis documentation\u003c/a\u003e."
+      ]
+    },
+    {
+      "name": "Tesla Model Y (with HW4) 2024",
+      "model": "Model Y (with HW4)",
+      "years": "2024",
+      "year_list": "2024",
+      "package": "All",
+      "angled_mount": false,
+      "video_link": "",
+      "harness_connector": "Tesla B",
+      "detail_sentence": "openpilot upgrades your <strong>Tesla Model Y (with HW4)</strong> with automated lane centering <strong>at all speeds</strong>, and adaptive cruise control <strong>that automatically resumes from a stop</strong>.",
```

Comment:
lol we might want to do a custom detail sentence for Tesla at some point :P

Reply:
or just all. what were you thinking?